### DISCUSSION TITLE
Performance of multiple count queries

### DISCUSSION THREAD

**Discussion by cupcakearmy** [prisma/prisma#15321] [Q&A]
Hello there!

What is the most efficient way to make a batch query for count queries? So basically instead of doing multiple `count()` is there a way to do that in a batch?

Thanks!

**Comment by nurul3101** [comment]
Hey @cupcakearmy 👋 
Did you try using [transactions](https://www.prisma.io/docs/guides/performance-and-optimization/prisma-client-transactions-guide) to batch all the count queries? Using transaction should be efficient in this use case.

**Reply by cupcakearmy** [reply]
Thanks for getting back!
In my (very limited) testing I found that manually calling multiple prisma `count` queries in an `Promise.All` is quite faster than putting them into a `Transaction`. I guess this makes sense as transaction is atomic according to the docs.
However is the roundtrip to the db the same?

**Reply by pantharshit00** [reply]
Hey @cupcakearmy, Did you use the interactive transaction API here (ie `.$transaction((tx)=> {})`) or the `.$transaction([])` API here? The first one be slower as it will store some state about the connection.

**Reply by cupcakearmy** [reply]
Quite sure I used the `[]` array based one of promises.

**Reply by pantharshit00** [reply]
Interesting, can you please report your findings here if you have some time: https://github.com/prisma/prisma/issues/new?assignees=&labels=kind%2Fbug&template=bug_report.yml

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
